packages/apps/human-app/frontend/src/api/services/worker/my-jobs-data.ts

```diff
@@ -9,12 +9,44 @@ import { createPaginationSchema } from '@/shared/helpers/create-pagination-schem
 import type { MyJobsFilterStoreProps } from '@/hooks/use-my-jobs-filter-store';
 import { useMyJobsFilterStore } from '@/hooks/use-my-jobs-filter-store';
 
+export enum MyJobStatus {
+  ACTIVE = 'ACTIVE',
+  CANCELED = 'CANCELED',
+  COMPLETED = 'COMPLETED',
+  VALIDATION = 'VALIDATION',
+  EXPIRED = 'EXPIRED',
+  REJECTED = 'REJECTED',
+}
+
+// Repeated because of linting rule
+export enum MyJobStatusWithUnknown {
+  ACTIVE = 'ACTIVE',
+  CANCELED = 'CANCELED',
+  COMPLETED = 'COMPLETED',
+  VALIDATION = 'VALIDATION',
+  EXPIRED = 'EXPIRED',
+  REJECTED = 'REJECTED',
+  UNKNOWN = 'UNKNOWN',
+}
+
 const myJobSchema = z.object({
   assignment_id: z.string(),
   escrow_address: z.string(),
   chain_id: z.number(),
   job_type: z.string(),
-  status: z.string(),
+  status: z.string().transform((value) => {
+    try {
+      return z
+        .enum([
```

Comment:
Shouldn't it be `.nativeEnum(MyJobStatus)`?
From what I see we can have at least `expired` jobs, so I assume all of them needed.